### ISSUE TITLE
Stuck in "Searching for devices..." on macOS 12.2 for Keychron Q1

### ISSUE THREAD

**Issue by pmatias** [the-via/releases#151]
What the title says. I've installed latest version of VIA, 1.3.1 on macOS 12.2.

The keyboard is connected directly into the USB port, no hubs involved. Interestingly enough, the "Key tester" section detects key presses.

Any help would be appreciated :)

**Comment by drahamim** [comment]
Seeing the same behavior. 
ENV:
MacOS 11.3.1
Via 1.3.1
Keyboard KBD75

**Comment by Olivier-tl** [comment]
Here are the instructions from Keychron : https://www.keychron.com/blogs/archived/how-to-use-via-to-program-your-keyboard

You need to download the keymap JSON file available on the [Q1 product page](https://www.keychron.com/products/keychron-q1-qmk-custom-mechanical-keyboard-knob-version).

At least this did the trick for my Q2 on macOS 12.01 :)

**Comment by Olivier-tl** [comment]
For the Q1, the instructions from keychron are [here](https://www.keychron.com/blogs/archived/how-to-use-via-to-program-your-keyboard) (you need to download the keymap JSON file from the keyboard product page and then load it into VIA).

**Comment by jlucier** [comment]
I followed those instructions exactly and was still seeing the behavior described by the OP. I did this in both Mac and Ubuntu 20.04, same result.

**Comment by Auroray** [comment]
Same issue here. Tried adding all kinds of permissions to VIA but the Q1 is not detected. 
OS: macOS 12.2
HW: MacBook Pro (15, 2018)
VIA v. 1.3.1
KB: Keychron Q1 QMK

**Comment by usagitchi** [comment]
Same issue, but on a Portico
OS: macOS 12.1
HW: MacBook Pro (13, 2015)
VIA v 1.3.1
KB: Portico

My keyboard actually stopped working for a while until I was able to plug it into a PC and get most of it mapped. This app still doesn't work for my keyboard though and I don't know why.

**Comment by purpleonyx** [comment]
Same issue, but on KBDFans DZ60
OS: macOS 12.2
HW: MacBook Pro (14, 2021)
VIA v 1.3.1
KB: DZ60RGB v2 Custom

**Comment by gbonanome** [comment]
Same issue also for me, tested on 2 different MacBook Pro: 
OS + HW: 
macOS 10.15.7 on MacBook Pro (13, early 2013)
and macOS 12.2 on MacBook Pro (14, 2021)
VIA v 1.3.1
KB: Keychron Q1 QMK ISO Knob version

In both cases the keyboard is not detected and everytime I open VIA I have to load the provided json.
Also, the tab LIGHTING crash the application, it show me an empty blue screen and I cannot go back or do anything.

**Comment by NhutLe2123** [comment]
Same issue for me on Mac pro M1 16 inch with K8 pro. I'm stuck at Screen "Searching for devices".
OS: Monterey 12.5 (Macbook Pro M1 - 16 inch - 2021)
VIA: v1.3.1
Keyboard: Keychron K8 Pro

**Comment by ocaro** [comment]
I am getting the same "Searching for devices..." with a Keychron Q1 V2, but able to configure the keyboard at www.usevia.app using Chrome Version 107.0.5304.121 (Official Build) (x86_64).
Mac mini 2018 (Intel)
OS: macOS Ventura 13.0.1
VIA Version 2.0.5 (2.0.5)

**Comment by jlucier** [comment]
We're at almost exactly 1 year from this being opened, and I think I got it working. I did this on linux using the native VIA AppImage, but hopefully this helps others too.

I got it working by loosely following these steps: https://www.keychron.com/pages/how-to-use-via-to-pair-with-keychron-q-series-keyboard

1. Download the json for your correct keyboard
2. Go to settings and enable "Show design tab"
3. In the design tab, **make sure you enable V2 definitions**
4. Then the keyboard will finally pair and become programmable

**Comment by praecipitator** [comment]
No. Tried with a keychron v6 on linux mint. With V2 definitions enabled, the JSON won't even import. 
The key tester works, though. 

Edit: I found a way to make it work. Not sure if this is user error or a bug, but it works the same both on windows and linux for me:

- Each time you start VIA, go to the Design tab
- Disregard any previously-imported designs, click "load" and select your json file.
- Switch to Configure, it should now work

Unless there is some super-hidden button to actually save and use the imported definition, it *must* be imported anew every single time you start VIA.

This is where to get keychron jsons. Make sure to not confuse the Max and the not-Max versions.
https://www.keychron.com/pages/firmware-and-json-files-of-the-keychron-qmk-v-and-v-max-series-keyboards
